src/entity/chaser.rs

```diff
@@ -0,0 +1,104 @@
+use crate::primitives::vector::Vector3;
+use crate::{server::server_state::PlayerState, world::World};
+use crate::primitives::position::{self, Position};
+use super::monster::{MonsterAction, TransitionState};
+
+const CHASING_DISTANCE: f32 = 10.;
+
+/// Internal state of the monster,
+/// name are for convinience, they are not force to a particular action
+#[derive(Clone)]
+pub enum MonsterStateEnum {
+    Idle,
+    Forward,
+    TurnLeft,
+    TurnRight,
+    Attack
+}
+
+pub struct Chaser {
+    state: MonsterStateEnum,
+    target: Vector3
+}
+
+impl TransitionState for Chaser {
+    fn new() -> Self {
+        Self {
+            state: MonsterStateEnum::Idle,
+            target: Vector3::empty()
+        }
+    }
+    fn action(&self) -> MonsterAction {
+        match self.state {
+            MonsterStateEnum::Forward => MonsterAction::Forward,
+            MonsterStateEnum::TurnLeft => MonsterAction::LeftRot,
+            MonsterStateEnum::TurnRight => MonsterAction::RightRot,
+            MonsterStateEnum::Attack => MonsterAction::Attack,
+            _ => MonsterAction::Idle
+            
+        }
+    }
+
+    fn update(&mut self, dt: f32, position: &Position, world: &World, player_list: Vec<PlayerState>) {
+        if player_list.len() > 0 {
+            // Try to find the closest player
+            let mut index_closest: Option<usize> = Option::None;
+            let mut closest_norm = 10.;
+            let mut norm = 0.;
+            for i in 0..player_list.len() {
+                norm = (player_list[i].pos.pos() - position.pos()).norm();
+                if  norm < CHASING_DISTANCE && norm < closest_norm {
+                    closest_norm = norm;
+                    index_closest = Some(i);
+                }
+            }
+            match index_closest {
+                    // A player is close enought, target it
+                Some(index) => {
+                    self.target = player_list[index].pos.pos();
+                    self.go_to_target(position)
+                },
+                // No player is close enought
+                _ => self.state = MonsterStateEnum::Idle
+            }
+        }
```

Comment:
This is really really performance intensive ... We should hold it into a state !